packages/core/types/src/http/product/admin/payloads.ts

```diff
@@ -210,10 +210,14 @@ export interface AdminCreateProduct {
    * The ID of the product's collection.
    */
   collection_id?: string
+  /**
+   * The ID of the product's shipping profile.
+   */
+  shipping_profile_id: string
```

Comment:
Q: I ve seen that it is handled as optional in the create product workflow, is this intentional to be required from the API?

Reply:
good catch, should be [mandatory](https://github.com/medusajs/medusa/pull/10902#discussion_r1922395161) in API and the workflow